storage/storage.go

```diff
@@ -1411,6 +1423,16 @@ func (q *Query) SetAttrSelection(attrs []string) error {
 	return nil
 }
 
+// GetProjection returns either value of Projection field or
+// default projection when this field is empty.
+func (q *Query) GetProjection() string {
```

Comment:
No need for a getter function here; you can just put this logic in ObjectIterator.fetch directly.